pandas/tests/io/json/test_pandas.py

```diff
@@ -1204,9 +1204,10 @@ def test_data_frame_size_after_to_json(self):
 
     @pytest.mark.parametrize('index', [None, [1, 2], [1., 2.], ['a', 'b'],
                                        ['1', '2'], ['1.', '2.']])
-    @pytest.mark.parametrize('columns', [['a', 'b'], ['1', '2'], ['1.', '2.']])
+    @pytest.mark.parametrize('columns', [None, [1, 2], [1., 2.], ['a', 'b'],
```

Comment:
So I don't know that we want to do this. Is it valid JSON in the table spec to have column names that are non-string?

Understood you have gotten this to round trip but if it violates the Table spec for JSON then I'd rather raise as commented previously

Reply:
@WillAyd as I already commented in #19129, after having tried to start a discussion before doing the PR:

> I was suggesting not just "raising a more descriptive ValueError" (sic), but changing the implementation of the JSON serialization for orient='table'.

Could you please tell me where the JSON table spec claims that column names MUST be strings?

I am going to make a longer comment to further justify my PR.

Reply:
> Could you please tell me where the JSON table spec claims that column names MUST be strings?

Not specific to the table spec as much as just JSON itself. See the description of an object here:

https://json.org

Reply:
@WillAyd you are talking about different things:
- JSON spec imposes in a JSON object (composed of key-value pairs), its keys must be strings
- but we are talking about column names, not JSON object keys

And my question is: where the JSON table spec claims that COLUMN NAMES (not JSON object keys) must be strings?

Reply:
@pwalsh can you comment on this? Is the `name` field in a Field Descriptor expected to be a string?